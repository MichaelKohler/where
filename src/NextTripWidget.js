import React from 'react';
import PropTypes from 'prop-types';
import './scss/widget.scss';

class NextTripWidget extends React.Component {
  render() {
    const { nextTrip, color } = this.props;
    let sectionClasses = 'card next';

    if (color) {
      sectionClasses += ` ${color}`;
    }

    return (
      <section className={sectionClasses}>
        <h1>
          Next trip:
          &nbsp;
          {nextTrip.destination}
        </h1>
        <p>
          {new Date(nextTrip.dateFrom).toDateString()}
          &nbsp;-&nbsp;
          {new Date(nextTrip.dateUntil).toDateString()}
        </p>
        { nextTrip.description ? <p>{nextTrip.description}</p> : '' }
      </section>
    );
  }
}

NextTripWidget.defaultProps = {
  nextTrip: {},
  color: ''
};

NextTripWidget.propTypes = {
  nextTrip: PropTypes.shape({
    destination: PropTypes.string,
    dateFrom: PropTypes.string,
    dateUntil: PropTypes.string,
    description: PropTypes.string
  }),
  color: PropTypes.string,
};

export default NextTripWidget;
