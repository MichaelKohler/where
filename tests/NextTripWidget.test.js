import React from 'react';
import { render } from '@testing-library/react';

import NextTripWidget from '../src/NextTripWidget';

test('should correctly render NextTripWidget without description', () => {
  const nextTrip = {
    destination: 'Bern',
    dateFrom: '2020-02-02',
    dateUntil: '2020-02-03',
  };
  const { container } = render(<NextTripWidget nextTrip={nextTrip} color="blue" />);
  expect(container.firstChild).toMatchSnapshot();
});

test('should correctly render NextTripWidget with description', () => {
  const nextTrip = {
    destination: 'Bern',
    dateFrom: '2020-02-02',
    dateUntil: '2020-02-03',
    description: 'Foo',
  };
  const { container } = render(<NextTripWidget nextTrip={nextTrip} color="blue" />);
  expect(container.firstChild).toMatchSnapshot();
});

test('should correctly render NextTripWidget empty if no data', () => {
  const { container } = render(<NextTripWidget />);
  expect(container.firstChild).toMatchSnapshot();
});
