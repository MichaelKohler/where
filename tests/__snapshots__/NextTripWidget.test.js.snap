// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render NextTripWidget empty if no data 1`] = ``;

exports[`should correctly render NextTripWidget with description 1`] = `
<section
  class="card next blue"
>
  <h1>
    Next trip:  
    Bern
  </h1>
  <p>
    Sun Feb 02 2020
     - 
    Mon Feb 03 2020
  </p>
  <p>
    Foo
  </p>
</section>
`;

exports[`should correctly render NextTripWidget without description 1`] = `
<section
  class="card next blue"
>
  <h1>
    Next trip:  
    Bern
  </h1>
  <p>
    Sun Feb 02 2020
     - 
    Mon Feb 03 2020
  </p>
  
</section>
`;
